homeassistant/components/uptimerobot/binary_sensor.py

```diff
@@ -21,69 +32,91 @@
 
-    up_robot = UptimeRobot()
-    api_key = config.get(CONF_API_KEY)
-    monitors = up_robot.getMonitors(api_key)
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="uptimerobot",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=60),
+    )
 
-    devices = []
-    if not monitors or monitors.get("stat") != "ok":
+    await coordinator.async_refresh()
+
+    if not coordinator.data or coordinator.data.get("stat") != "ok":
         _LOGGER.error("Error connecting to Uptime Robot")
-        return
+        raise PlatformNotReady()
 
-    for monitor in monitors["monitors"]:
-        devices.append(
+    async_add_entities(
+        [
             UptimeRobotBinarySensor(
-                api_key,
-                up_robot,
-                monitor["id"],
-                monitor["friendly_name"],
-                monitor["url"],
+                coordinator,
+                UptimeRobotBinarySensorEntityDescription(
+                    key=monitor["id"],
+                    name=monitor["friendly_name"],
+                    target=monitor["url"],
+                    device_class=DEVICE_CLASS_CONNECTIVITY,
+                ),
             )
-        )
-
-    add_entities(devices, True)
+            for monitor in coordinator.data["monitors"]
+        ],
+        True,
+    )
 
 
-class UptimeRobotBinarySensor(BinarySensorEntity):
+class UptimeRobotBinarySensor(BinarySensorEntity, CoordinatorEntity):
     """Representation of a Uptime Robot binary sensor."""
 
-    def __init__(self, api_key, up_robot, monitor_id, name, target):
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        description: UptimeRobotBinarySensorEntityDescription,
+    ) -> None:
         """Initialize Uptime Robot the binary sensor."""
-        self._api_key = api_key
-        self._monitor_id = str(monitor_id)
-        self._name = name
-        self._target = target
-        self._up_robot = up_robot
-        self._state = None
-
-    @property
-    def name(self):
-        """Return the name of the binary sensor."""
-        return self._name
-
-    @property
-    def is_on(self):
-        """Return the state of the binary sensor."""
-        return self._state
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return DEVICE_CLASS_CONNECTIVITY
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes of the binary sensor."""
-        return {ATTR_ATTRIBUTION: ATTRIBUTION, ATTR_TARGET: self._target}
-
-    def update(self):
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self.entity_description = description
+        self._attr_extra_state_attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_TARGET: self.entity_description.target,
+        }
+
+    async def async_update(self):
```

Comment:
Remove `async_update` and implement `_handle_coordinator_update` callback method or just the `is_on` property.